dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -466,6 +472,13 @@ public JdbcEventStore( StatementBuilder statementBuilder, JdbcTemplate jdbcTempl
                 event.getNotes().add( note );
                 notes.add( rowSet.getString( "psinote_id" ) );
             }
+
+            List<Long> relationshipIds = jdbcTemplate.queryForList( RELATIONSHIP_QUERY + rowSet.getLong( "psi_id" ),
```

Comment:
I think this wont scale well. I think the original idea of integrating this into the original might be better. What was the challenges of doing this?

Reply:
We are basically doing 2N queries on the same tables, when we could potentially do it in 1 go

Reply:
resolved